### PR TITLE
tree-sitter 0.24 upgrade

### PR DESCRIPTION
**Pull request by mwhitworth**
needs #279 

changes required to upgrade to tree-sitter 0.24. But it looks like there is at least one issue with a 0.24 upgrade (#282), and upgrading to 0.23 has the advantage of moving Rust bindings to `tree-sitter-language` and simplifying further upgrades there.

### CODE REVIEW

**Comment by matthias-Q** [comment]
@mwhitworth can you rebase `main` into your branch please? I have updated CI and also bumped tree-sitter to 0.23. CI should pass with tree-sitter 0.24 as well.

Once this lands I would make a new release.

**Comment by mwhitworth** [comment]
I'm rebasing this now (and checking the diff)

I came across a message while cloning the repository, and some odd behaviour around `bindings/swift/TreeSitterSQL/sql.h`. This is because there are actually two copies of the file in the working tree. Running `git clone` on the repo displays the following message:

```
warning: the following paths have collided (e.g. case-sensitive paths
on a case-insensitive filesystem) and only one from the same
colliding group is in the working tree
  'bindings/swift/TreeSitterSQL/sql.h'
  'bindings/swift/TreeSitterSql/sql.h'
```

It might be clearest to fix this separately.

File: Cargo.toml [hunk 1]
```diff
@@ -23,7 +21,7 @@ path = "bindings/rust/lib.rs"
 tree-sitter-language = "0.1"
 
 [build-dependencies]
-cc = "~1.2.1"
```
Comment (matthias-Q) [inline Cargo.toml:26]: I think we should go with ~1.2.1

File: bindings/rust/build.rs [hunk 2]
```diff
@@ -2,23 +2,21 @@ fn main() {
...
     #[cfg(target_env = "msvc")]
     c_config.flag("-utf-8");
 
     let parser_path = src_dir.join("parser.c");
     c_config.file(&parser_path);
+    println!("cargo:rerun-if-changed={}", parser_path.to_str().unwrap());
 
-    // If your language uses an external scanner written in C,
-    // then include this block of code:
+    // NOTE: if your language uses an external scanner, uncomment this block:
```
Comment (matthias-Q) [inline bindings/rust/build.rs:14]: I think we should do this. But maybe it would be also a separate PR

**Review by matthias-Q — CHANGES_REQUESTED** [review]
LGTM. Can you make these two changes?

**Comment by mwhitworth** [comment]
> ....
> It might be clearest to fix this separately.

https://github.com/DerekStride/tree-sitter-sql/pull/290

Then I'll finalize this branch for merge.

File: bindings/rust/build.rs (hunk 2 above)
Comment (mwhitworth) [reply bindings/rust/build.rs:14]: I'll fix this in this branch.

(It also makes me curious about making the upgrades easier in `tree-sitter` itself, given that e.g. filesystem state should be easy to inspect during upgrades)

File: Package.swift [hunk 3]
```diff
@@ -1,34 +1,37 @@
 // swift-tools-version:5.3
-
 import PackageDescription
 
 let package = Package(
-    name: "TreeSitterSQL",
-    platforms: [.macOS(.v10_13), .iOS(.v11)],
+    name: "TreeSitterSql",
```
Comment (mwhitworth) [inline Package.swift:5]: Two meaningful changes to note:
- the Swift package name has been changed - I don't know if capitalization matters for this package manager
- the `exclude` paths have disappeared, but the effect on the package should be minimal.

**Comment by mwhitworth** [comment]
With this rebase and some further `fix` commits, this should be good to go. Ideally there should be less `fix` commits for each `tree-sitter` release, but I suspect this is mostly an upstream issue. Since I've helped to apply `tree-sitter` upgrades across some other language grammars, I'll be paying attention to this.

**Review by matthias-Q — APPROVED** [review]

**Comment by matthias-Q** [comment]
Thanks @mwhitworth for the effort!
